### PR TITLE
Add tests for well-known Symbol: @@toPrimitive

### PR DESCRIPTION
**Pull request by jugglinmike**

### CODE REVIEW

File: test/built-ins/Date/prototype/Symbol.toPrimitive/hint-default.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+assert.sameValue(voCallCount, 1, '`valueOf` method was invoked exactly once');
+
+obj = {
+  valueOf: null,
+  toString: null
+};
+
+assert.throws(TypeError, function() {
+  Date.prototype[Symbol.toPrimitive].call(obj, 'default');
+});
```
Comment (rwaldron) [inline test/built-ins/Date/prototype/Symbol.toPrimitive/hint-default.js]: This file is pretty heavy—any thoughts on breaking it up?

File: test/built-ins/Date/prototype/Symbol.toPrimitive/hint-number.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+assert.sameValue(tsCallCount, 1, '`toString` method was invoked exactly once');
+
+obj = {
+  valueOf: null,
+  toString: null
+};
+
+assert.throws(TypeError, function() {
+  Date.prototype[Symbol.toPrimitive].call(obj, 'number');
+});
```
Comment (rwaldron) [inline test/built-ins/Date/prototype/Symbol.toPrimitive/hint-number.js]: Same question, re: breakup

File: test/built-ins/Date/prototype/Symbol.toPrimitive/hint-string.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+assert.sameValue(voCallCount, 1, '`valueOf` method was invoked exactly once');
+
+obj = {
+  valueOf: null,
+  toString: null
+};
+
+assert.throws(TypeError, function() {
+  Date.prototype[Symbol.toPrimitive].call(obj, 'string');
+});
```
Comment (rwaldron) [inline test/built-ins/Date/prototype/Symbol.toPrimitive/hint-string.js]: Same question, re: breakup

**Comment by rwaldron** [comment]
Aside from my questions about the files that have a lot of things in them, this looks good to me.

**Comment by jugglinmike** [comment]
@rwaldron Thanks for the review! I've split the three files you mentioned into 4 parts each: `*-first-valid.js`, `*-first-invalid.js`, `*-first-non-callable.js`, and `*-no-callables.js`. What do you think?

File: test/built-ins/Date/prototype/Symbol.toPrimitive/hint-default-first-invalid.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  }
+};
+
+assert.sameValue(
+  Date.prototype[Symbol.toPrimitive].call(obj, 'default'),
+  'valueOf test262',
+  '`valueOf` is used as a fallback when `toString` returns an object'
+);
+assert.sameValue(voCallCount, 1, '`valueOf` method was invoked exactly once');
+assert.sameValue(tsCallCount, 1, '`toString` method was invoked exactly once');
```
Comment (farzonl) [inline test/built-ins/Date/prototype/Symbol.toPrimitive/hint-default-first-invalid.js]: Help me understand this, we call toPrimitive once but some how we invoke both valueOf and toString? seems to me default flag should default to either toString or valueOf not invoke both

File: test/built-ins/Date/prototype/Symbol.toPrimitive/hint-invalid.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+assert.throws(TypeError, function() {
+  d[Symbol.toPrimitive]('String');
+});
+
+assert.throws(TypeError, function() {
+  d[Symbol.toPrimitive]('defaultnumber');
+});
+
+assert.throws(TypeError, function() {
+  d[Symbol.toPrimitive](new String('number'));
```
Comment (farzonl) [inline test/built-ins/Date/prototype/Symbol.toPrimitive/hint-invalid.js]: //TODO review with spec

File: test/built-ins/Date/prototype/Symbol.toPrimitive/hint-invalid.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+});
+
+assert.throws(TypeError, function() {
```
Comment (farzonl) [inline test/built-ins/Date/prototype/Symbol.toPrimitive/hint-invalid.js]: //TODO review with spec

**Comment by farzonl** [comment]
LGTM

**Comment by goyakin** [comment]
Thanks @jugglinmike.

**Comment by rwaldron** [comment]
> What do you think?

I was on vacation, but I definitely approve :D

**Pull request merged**